src/workerd/jsg/fast-api.h

```diff
@@ -0,0 +1,131 @@
+#pragma once
+
+#include <workerd/jsg/web-idl.h>
+
+#include <v8-fast-api-calls.h>
+#include <v8-local-handle.h>
+#include <v8-value.h>
+
+#include <kj/common.h>
+
+namespace workerd::jsg {
+
+template <typename T>
+struct IsFastApiCompatibleWrappedType1 {
+  static constexpr bool value = webidl::isDictionaryType<T> ||
+      webidl::isNonCallbackInterfaceType<T> || kj::isSameType<T, kj::String>() ||
+      kj::isSameType<T, jsg::BufferSource>();
+};
+
+template <typename T>
+struct IsFastApiMaybe {
+  static constexpr bool value = false;
+};
+
+template <typename U>
+struct IsFastApiMaybe<kj::Maybe<U>> {
+  static constexpr bool value = IsFastApiCompatibleWrappedType1<U>::value;
+};
+
+template <typename U>
+struct IsFastApiMaybe<jsg::Optional<U>> {
+  static constexpr bool value = IsFastApiCompatibleWrappedType1<U>::value;
+};
+
+template <typename U>
+struct IsFastApiMaybe<jsg::LenientOptional<U>> {
+  static constexpr bool value = IsFastApiCompatibleWrappedType1<U>::value;
+};
+
+template <typename T>
+struct IsFastApiCompatibleWrappedType {
+  static constexpr bool value =
+      IsFastApiCompatibleWrappedType1<T>::value || IsFastApiMaybe<T>::value;
+};
+
+// Helper to determine if a type can be used as a return type in V8 Fast API
+template <typename T>
+struct IsFastApiCompatibleReturnType {
+  static constexpr bool value = kj::isSameType<T, void>() || kj::isSameType<T, bool>() ||
```

Comment:
It seems like we cannot return wrapped types, etc. Is that a limitation of the fast API, or just not yet covered in this PR?

Reply:
Current limitation of the fast API. While it now accepts a broader range of input arguments it is strictly limited in the types of return values it can handle

Reply:
Darn